fs/ftpfs.py

```diff
@@ -212,20 +222,22 @@ def write(self, data):
                 remaining_data -= sent_size
                 self.pos += sent_size
 
+        return data_pos
+
     def writelines(self, lines):
         self.write(b''.join(lines))
 
     def truncate(self, size=None):
         # Inefficient, but I don't know if truncate is possible with ftp
         with self._lock:
-            if size is None:
-                size = self.tell()
+            size = size or self.tell()
```

Comment:
I think this would break if you did `truncate(0)`. Would it not truncate at the current file position?

Reply:
Yes you're right, I'll change it to `self.tell() if size is None else size`.